src/System/TimedScopes/TimedScope.cs

```diff
@@ -28,20 +27,23 @@ public class TimedScope : IDisposable
 		/// <param name="scopeLogger">Scope metrics logger</param>
 		/// <param name="replayEventConfigurator">Replay event configurator</param>
 		/// <param name="machineInformation">Machine Information</param>
+		/// <param name="timedScopeStackManager">Timed scope stack manager</param>
 		/// <param name="correlationData">Correlation data</param>
-		private TimedScope(TimedScopeDefinition scopeDefinition, CorrelationData correlationData, ITimedScopeLogger scopeLogger, 
-			IReplayEventConfigurator replayEventConfigurator, IMachineInformation machineInformation)
+		private TimedScope(TimedScopeDefinition scopeDefinition, CorrelationData correlationData, ITimedScopeLogger scopeLogger, IReplayEventConfigurator replayEventConfigurator,
+			IMachineInformation machineInformation, ITimedScopeStackManager timedScopeStackManager)
 		{
 
 			Code.ExpectsArgument(scopeDefinition, nameof(scopeDefinition), TaggingUtilities.ReserveTag(0x238174da /* tag_96xt0 */));
 			Code.ExpectsArgument(scopeLogger, nameof(scopeLogger), TaggingUtilities.ReserveTag(0x238174db /* tag_96xt1 */));
 			Code.ExpectsArgument(replayEventConfigurator, nameof(replayEventConfigurator), TaggingUtilities.ReserveTag(0x238174dc /* tag_96xt2 */));
+			Code.ExpectsArgument(timedScopeStackManager, nameof(timedScopeStackManager), TaggingUtilities.ReserveTag(0));
```

Comment:
>timedScopeStackManager [](start = 55, length = 22)

The timed scope manager is null by default, wouldn't that fail the constructor here?

Reply:
I kept the old logic in constructor; scopeLoger which was the original parameter also is by default null and still is checked; the same I applied to the other parameters (however it does not really make much sense to me).

Reply:
Can you please make sure it works as expected.

---
In reply to: [314390077](https://github.com/microsoft/Omex/pull/96#discussion_r314390077) [](ancestors = 314390077)

Reply:
As the timed scope manager is obligatory for times scope to work, I made it an obligatory parameter. The same applies to other parameters: scopeLogger, replayEventConfigurator.